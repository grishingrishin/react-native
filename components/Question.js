import React from 'react';
import { TouchableOpacity } from 'react-native';
import { Link } from 'react-router-native';
import styled from 'styled-components/native';

const Container = styled.View`
  flex-direction: row;
  justify-content: center;
  margin-top: 18px;
  text-align: center;
`;

const Question = styled.Text`
  font-size: 12px;
  font-family: Roboto-Thin;
  color: #fff;
`;

const Route = styled.Text`
  text-decoration: underline;
  font-size: 12px;
  font-family: Roboto-Regular;
  color: #fff;
`;

export default ({ question, path, route }) => (
  <Container>
    <Question>{question} </Question>
    <Link to={path} component={TouchableOpacity}>
      <Route>{route}</Route>
    </Link>
  </Container>
);