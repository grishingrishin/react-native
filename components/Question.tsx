import React from 'react';
import {TouchableOpacity} from 'react-native';
import {Link} from 'react-router-native';
import styled from 'styled-components/native';

const Container = styled.View`
  flex-direction: row;
  justify-content: center;
  margin-top: 18px;
  text-align: center;
`;

const Question = styled.Text`
  font-size: 12px;
  font-family: Roboto-Thin;
  color: #fff;
`;

const Route = styled.Text`
  text-decoration: underline;
  font-size: 12px;
  font-family: Roboto-Regular;
  color: #fff;
`;

interface QProps {
  question: string;
  route: string;
  path: string;
}

export default ({question, route, path}: QProps) => (
  <Container>
    <Question>{question} </Question>
    <Link to={path} component={TouchableOpacity}>
      <Route>{route}</Route>
    </Link>
  </Container>
);
