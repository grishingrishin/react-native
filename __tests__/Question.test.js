import React from 'react';
import { shallow } from 'enzyme';
import { create } from 'react-test-renderer';
import Question from '../components/Question';

const props = {
  question: 'Some test text question',
  route: 'Some test text route',
  path: '/some_test_path',
}

describe('<Question />', () => {
  describe('Jest tests', () => {
    it('renders correctly without props', () => {
      const tree = create(<Question />).toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('renders correctly with props', () => {
      const tree = create(
        <Question 
          {...props}
        />
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  describe('Enzyme test', () => {
    it('allow us to set props and examp it value in nodes', () => {
      const wrapper = shallow(
        <Question
          {...props}
        />
      );

      const questionText = wrapper.find('Question').first();
      expect(questionText.text()).toEqual(props.question + ' ');  // Space important

      const routeText = wrapper.find('Question__Route').first();
      expect(routeText.text()).toEqual(props.route);

      expect(wrapper.find('Link').props().to).toEqual(props.path);
    });
  });
});
