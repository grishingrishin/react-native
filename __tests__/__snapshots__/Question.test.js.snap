// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Question /> Jest tests renders correctly with props 1`] = `
<View
  style={
    Array [
      Object {
        "flexDirection": "row",
        "justifyContent": "center",
        "marginTop": 18,
        "textAlign": "center",
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#fff",
          "fontFamily": "Roboto-Thin",
          "fontSize": 12,
        },
      ]
    }
  >
    Some test text question
     
  </Text>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#fff",
            "fontFamily": "Roboto-Regular",
            "fontSize": 12,
            "textDecorationColor": "black",
            "textDecorationLine": "underline",
            "textDecorationStyle": "solid",
          },
        ]
      }
    >
      Some test text route
    </Text>
  </View>
</View>
`;

exports[`<Question /> Jest tests renders correctly without props 1`] = `
<View
  style={
    Array [
      Object {
        "flexDirection": "row",
        "justifyContent": "center",
        "marginTop": 18,
        "textAlign": "center",
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#fff",
          "fontFamily": "Roboto-Thin",
          "fontSize": 12,
        },
      ]
    }
  >
     
  </Text>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#fff",
            "fontFamily": "Roboto-Regular",
            "fontSize": 12,
            "textDecorationColor": "black",
            "textDecorationLine": "underline",
            "textDecorationStyle": "solid",
          },
        ]
      }
    />
  </View>
</View>
`;
